test/unit/org/apache/cassandra/cql3/validation/entities/SecondaryIndexTest.java

```diff
@@ -1052,6 +1055,104 @@ public void testIndexQueriesWithIndexNotReady() throws Throwable
+            String tableName = createTable("CREATE TABLE %s (pk int, ck int, value int, PRIMARY KEY (pk, ck))");
+            String indexName = createIndex("CREATE CUSTOM INDEX ON %s (value) USING '" + LoadTypeConstrainedIndex.class.getName() + "'");
+            assertTrue(waitForIndex(keyspace(), tableName, indexName));
+            execute("SELECT value FROM %s WHERE value = 1");
+            execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 1, 1);
+            index = (LoadTypeConstrainedIndex) getCurrentColumnFamilyStore().indexManager.getIndexByName(indexName);
+            assertEquals(0, index.rowsInserted.size());
+
+            // Upon recovery, we can index data again
+            index.reset();
+            getCurrentColumnFamilyStore().indexManager.rebuildIndexesBlocking(ImmutableSet.of(indexName));
+            execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 2, 1, 1);
+            assertEquals(1, index.rowsInserted.size());
+            execute("DROP index " + KEYSPACE + "." + indexName);
+
+            // On bad init writes are not forwarded to the index
+            index.reset();
+            LoadTypeConstrainedIndex.setSupportedLoad(LoadType.WRITE);
+            index.failInit = true;
+            indexName = createIndex("CREATE CUSTOM INDEX ON %s (value) USING '" + LoadTypeConstrainedIndex.class.getName() + "'");
+            assertTrue(waitForIndexBuilds(keyspace(), indexName));
+            execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 1, 1);
+            assertEquals(0, index.rowsInserted.size());
+            execute("DROP index " + KEYSPACE + "." + indexName);
+        }
+        finally
+        {
+            if (index != null)
+                index.reset();
+        }
+    }
+    
+    @Test  // A Bad init could leave an index only accepting writes
+    public void testWriteOnlyIndex() throws Throwable
+    {
+        LoadTypeConstrainedIndex index = null;
+        try
+        {
+            // Writes go through but reads fail
+            LoadTypeConstrainedIndex.setSupportedLoad(LoadType.WRITE);
+            String tableName = createTable("CREATE TABLE %s (pk int, ck int, value int, PRIMARY KEY (pk, ck))");
+            String indexName = createIndex("CREATE CUSTOM INDEX ON %s (value) USING '" + LoadTypeConstrainedIndex.class.getName() + "'");
+            assertTrue(waitForIndex(keyspace(), tableName, indexName));
+            execute("INSERT INTO %s (pk, ck, value) VALUES (?, ?, ?)", 1, 1, 1);
+            index = (LoadTypeConstrainedIndex) getCurrentColumnFamilyStore().indexManager.getIndexByName(indexName);
+            assertEquals(1, index.rowsInserted.size());
+            try
+            {
+                execute("SELECT value FROM %s WHERE value = 1");
+                fail();
+            }
+            catch (IndexNotAvailableException e)
+            {
+                assertTrue(true);
+            }
```

Comment:
Nit: we can use the provided `assertInvalidThrow`:
```suggestion
            assertInvalidThrow(IndexNotAvailableException.class, "SELECT value FROM %s WHERE value = 1");
```